padrino-core/lib/padrino-core/path_router/compiler.rb

```diff
@@ -34,12 +34,23 @@ def compiled?
         !!@compiled
       end
   
+
+      ##
+      # Finds routes by using request or env.
+      #
+      def find_by(request_or_env)
+        request = request_or_env.is_a?(Hash) ? Sinatra::Request.new(request_or_env) : request_or_env
+        pattern = encode_default_external(request.path_info)
+        verb    = request.request_method.downcase.to_sym
```

Comment:
Should we normalise this bit [here too](https://github.com/padrino/padrino-framework/pull/1863/files#diff-7de007036e25d75ce0d909f152958551R56) and either extract `verb` everywhere or use `request.request_method.downcase.to_sym`?